### PR TITLE
Addressed multiple code comments and documentation issues

### PR DESCRIPTION
**Pull request by soninaren**

### CODE REVIEW

File: sdk/src/sdk/src/main/java/com/microsoft/windowsazure/mobileservices/ApiJsonOperationCallback.java [hunk 1]
```diff
@@ -34,7 +34,7 @@
     /**
      * Method to call when the operation is completed
      *
-     * @param jsonElement The obtained jsonElement
+     * @param jsonObject The obtained jsonElement
```
Comment (mamaso) [inline sdk/src/sdk/src/main/java/com/microsoft/windowsazure/mobileservices/ApiJsonOperationCallback.java]: if possible, I'd change the name of the parameter to jsonElement instead (as JsonObject is a type as well)

File: sdk/src/sdk/src/main/java/com/microsoft/windowsazure/mobileservices/MobileServiceClient.java [hunk 2]
```diff
@@ -1506,7 +1491,7 @@ public MobileServiceClient withFilter(final ServiceFilter serviceFilter) {
 
     /**
      * Gets the ServiceFilter. If there is no ServiceFilter, it creates and
-     * returns a default filter
+     * @return a default filter
```
Comment (mamaso) [inline sdk/src/sdk/src/main/java/com/microsoft/windowsazure/mobileservices/MobileServiceClient.java]: already has a @return statement

File: sdk/src/sdk/src/main/java/com/microsoft/windowsazure/mobileservices/MobileServiceClient.java [hunk 3]
```diff
@@ -964,7 +955,7 @@ public MobileServiceJsonTable getTable(String name) {
     }
 
     /**
-     * Returns a MobileServiceJsonSyncTable instance, which provides untyped
+     * @return a MobileServiceJsonSyncTable instance, which provides untyped
```
Comment (mamaso) [inline sdk/src/sdk/src/main/java/com/microsoft/windowsazure/mobileservices/MobileServiceClient.java]: duplicate @ return, seems like there are quite a few of these

**Comment by soninaren** [comment]
Went through all the @ return and fixed them

**Pull request merged**